src/autumnaton.ts

```diff
@@ -0,0 +1,189 @@
+  }
+}
+
+function seasonalItemValue(location: Location, seasonalOverride?: number): number {
+  // Find the value of the drops based on zone difficulty/type
+  const autumnItems = $items`autumn leaf, AutumnFest ale, autumn breeze, autumn dollar, autumn years wisdom`;
+  const avgValueOfRandomAutumnItem = garboAverageValue(...autumnItems);
+  const autumnMeltables = $items`autumn debris shield, autumn leaf pendant, autumn sweater-weather sweater`;
+  const autumnItem = AutumnAton.getUniques(location)?.item;
+  const seasonalItemDrops = seasonalOverride ?? AutumnAton.seasonalItems();
+  if (autumnItem) {
+    return (
+      (seasonalItemDrops > 1 ? avgValueOfRandomAutumnItem : 0) +
+      (autumnMeltables.includes(autumnItem)
+        ? // If we already have the meltable, then we get a random item, else value at 0
+          availableAmount(autumnItem) > 0
+          ? avgValueOfRandomAutumnItem
+          : 0
+        : garboValue(autumnItem, true))
+    );
+  } else {
+    // If we're in a location without any uniques, we still get cowcatcher items
+    return seasonalItemDrops > 1 ? avgValueOfRandomAutumnItem : 0;
+  }
+}
+
+function expectedRemainingExpeditions(legOverride?: number): number {
+  const availableAutumnatonTurns = estimatedTurns() - AutumnAton.turnsLeft();
+  let expeditionTurnSum = 0;
+  let quests = get("_autumnatonQuests", 0);
+  while (expeditionTurnSum < availableAutumnatonTurns) {
+    expeditionTurnSum +=
+      11 *
+      Math.max(
+        1,
+        quests -
+          (legOverride ?? AutumnAton.currentUpgrades().filter((u) => u.includes("leg")).length)
+      );
+    quests++;
+  }
+
+  return quests - get("_autumnatonQuests", 0);
+}
+
+const profitRelevantUpgrades = [
+  "leftarm1",
+  "leftleg1",
+  "rightarm1",
+  "rightleg1",
+  "cowcatcher",
+  "periscope",
+  "radardish",
+] as const;
+
+export function mostValuableUpgrade(fullLocations: Location[]): Location[] {
+  // This function shouldn't be getting called if we don't have an expedition left
+  if (expectedRemainingExpeditions() < 1) {
+    return fullLocations;
+  }
+  const currentUpgrades = AutumnAton.currentUpgrades();
+  const acquirableUpgrades = profitRelevantUpgrades.filter(
+    (upgrade) => !currentUpgrades.includes(upgrade)
+  );
+
+  if (acquirableUpgrades.length === 0) {
+    return fullLocations;
+  }
+
+  const currentBestLocation = maxBy(fullLocations, (loc: Location) => averageAutumnatonValue(loc));
+  const currentExpectedProfit =
+    averageAutumnatonValue(currentBestLocation) * expectedRemainingExpeditions();
+
+  const upgradeValuations = acquirableUpgrades.map((upgrade) => {
+    const upgradeLocations = fullLocations.filter(
+      (location) => AutumnAton.getUniques(location)?.upgrade === upgrade
+    );
+    const bestLocContainingUpg = maxBy(upgradeLocations, (loc: Location) =>
+      averageAutumnatonValue(loc)
+    );
+
+    if (upgrade === ("periscope" || "radardish")) {
```

Comment:
```suggestion
    if (["periscope", "radardish"].includes(upgrade)) {
```

The line as is does not do what you want it to do. It will see if upgrade is equal to `"periscope" || "radardish"`, which will evaluate to `"periscope"` (strings are falsy only when empty, and an "or" stops caring once it hits true)

Reply:
Also, it looks like you're separately considering several cases using a series of `if` statements; we can either use a `switch` or some sort of iterable to achieve this nicer. That is, you might have something like
```
switch (upgrade) {
  case "periscope":
  case "radardish": {
    all this stuff
  }
  case "rightleg1":
  case "rightleg2": {
    et cetera
  }
  case default:
    return DEFAULT_VALUE;
}
```
or something like
```
for (const { upgrades, profitFunction } of upgradeCases) {
  if (upgrades.includes(upgrade)) return { upgrade, profit: profitFunction() }
}
return DEFAULT_VALUE;
```
with better variable names than the ones I chose, and defining `upgradeCases` outside the scope of this function. We can wait to decide how exactly to set this up until after we've rejiggered the rest of it, though

Reply:
I was originally going to do the switch case but I ran into "no fall-through" lint errors before, and thought that this counted as such.

I will try doing it again